src/timely-util/src/builder_async.rs

```diff
@@ -295,14 +295,18 @@ where
     }
 }
 
-impl<'a, T, D, P> AsyncOutputHandle<T, CapacityContainerBuilder<Vec<D>>, P>
+impl<T, CB, P> AsyncOutputHandle<T, CB, P>
 where
     T: Timestamp,
-    D: Data,
-    P: Push<Bundle<T, Vec<D>>> + 'static,
+    CB: ContainerBuilder,
+    P: Push<Bundle<T, CB::Container>>,
 {
     #[allow(clippy::unused_async)]
-    pub async fn give<C: CapabilityTrait<T>>(&mut self, cap: &C, data: D) {
+    pub async fn give<C, D>(&mut self, cap: &C, data: D)
```

Comment:
You could use the same trick that timely uses: Implement `give` for a handle that is specialized to `CapacityContainerBuilder` and generic over a container, and add a separate `give_with_builder` that's implemented for handles generic over any container builder. This would remove the need for the second commit.

Reply:
Great idea! I actually only want `give` to be generic over the container so I'll skip adding a `give_with_builder` for now